src/static_schema.rs

```diff
@@ -176,3 +176,58 @@ fn default_dict_id() -> i64 {
 fn default_dict_is_ordered() -> bool {
     false
 }
+
+fn validate_field_names(field_name: &str, existing_fields: &mut HashSet<String>) -> Result<(), StaticSchemaError> {
+
+    if field_name.is_empty() {
+        return Err(StaticSchemaError::EmptyFieldName);
+    }
+
+    if !existing_fields.insert(field_name.to_string()) {
+        return Err(StaticSchemaError::DuplicateField(field_name.to_string()));
+    }
+
+    Ok(())
+}
+
+
+#[derive(Debug, thiserror::Error)]
+pub enum StaticSchemaError{
+
+    #[error(
+        "custom partition field {0} does not exist in the schema for the static schema logstream"
+    )]
+    MissingCustomPartition(String),
+
+    #[error(
+        "time partition field {0} does not exist in the schema for the static schema logstream"
+    )]
+    MissingTimePartition(String),
+
+    #[error("field {DEFAULT_TIMESTAMP_KEY:?} is a reserved field")]
+    DefaultTime,
```

Comment:
```suggestion
    DefaultTimestamp,
```

This should make more sense